test/e2e/serial/tests/non_regression.go

```diff
@@ -421,4 +422,115 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload placeme
+			By(fmt.Sprintf("creating a deployment %q with replicas %d candidate nodes %d", dpName, replicas, len(nrtCandidates)))
+
+			nroSchedObj := nrosched.CheckNROSchedulerAvailable(ctx, fxt.Client, objectnames.DefaultNUMAResourcesSchedulerCrName)
+
+			podLabels := map[string]string{
+				"test": dpName,
+			}
+
+			schedulerName := nroSchedObj.Status.SchedulerName // shortcut
+			podSpec := corev1.PodSpec{
+				SchedulerName: schedulerName,
+				Containers: []corev1.Container{
+					{
+						Name:  dpName + "-cnt",
+						Image: images.GetPauseImage(),
+						Command: []string{
+							images.PauseCommand,
+						},
+						Resources: corev1.ResourceRequirements{
+							Requests: requiredRes,
+							Limits:   requiredRes,
+						},
+					},
+				},
+				RestartPolicy: corev1.RestartPolicyAlways,
+			}
+			dp := objects.NewTestDeploymentWithPodSpec(replicas, podLabels, nodeSelector, fxt.Namespace.Name, dpName, podSpec)
+			Expect(fxt.Client.Create(ctx, dp)).To(Succeed())
+
+			// although the deployment pods will be pending thus the deployment will not be counted as complete,
+			// we need to wait until all the replicas are created despite their status before moving forward with the checks
+			By("wait for the deployment to be up with its pod created")
+			dp, err = wait.With(fxt.Client).Interval(time.Second).Timeout(time.Minute).ForDeploymentReplicasReadiness(ctx, dp, replicas)
+			Expect(err).NotTo(HaveOccurred())
+
+			By(fmt.Sprintf("checking deployment pods have been scheduled by %q ", schedulerName))
+			pods, err := podlist.With(fxt.Client).ByDeployment(ctx, *dp)
+			Expect(err).ToNot(HaveOccurred(), "unable to get pods from deployment %q:  %v", dp.Name, err)
+			Expect(pods).ToNot(BeEmpty(), "cannot find any pods for DP %s/%s", dp.Namespace, dp.Name)
+
+			var errorPods []string
+			for _, pod := range pods {
+				schedOK, err := nrosched.CheckPODWasScheduledWith(fxt.K8sClient, pod.Namespace, pod.Name, schedulerName)
+				if err != nil {
+					_ = objects.LogEventsForPod(fxt.K8sClient, pod.Namespace, pod.Name)
+				}
+				Expect(err).ToNot(HaveOccurred())
+				if !schedOK {
+					errorPods = append(errorPods, fmt.Sprintf("%s/%s", pod.Namespace, pod.Name))
+					klog.Warningf("pod %s/%s was NOT scheduled with %q", pod.Namespace, pod.Name, schedulerName)
+					continue
+				}
+			}
+			Expect(errorPods).To(BeEmpty(), "some pods not scheduled with %s", schedulerName)
+
+			replicas *= 2
+
+			By("updating deployment in such way each node will have a pod per NUMA zone")
+			Eventually(func() error {
+				err := fxt.Client.Get(ctx, client.ObjectKeyFromObject(dp), dp)
+				if err != nil {
+					return err
+				}
+				klog.Infof("upscaling replicas: %v -> %v", *dp.Spec.Replicas, replicas)
+				dp.Spec.Replicas = &replicas
+				return fxt.Client.Update(ctx, dp)
+			}).WithPolling(1*time.Second).WithTimeout(1*time.Minute).Should(Succeed(), "cannot downsize the test deployment %q", dp.Name)
+
+			By("waiting for some of the pods to be running")
```

Comment:
some? don't we expect all replicas to run

Reply:
yes, stale message. Fixed in the last upload.